test/jdk/java/awt/event/ComponentEvent/ComponentEventTest.java

```diff
@@ -0,0 +1,325 @@
+        choice.add("Red");
+        choice.add("Orange");
+        choice.add("Yellow");
+        Checkbox checkbox = new Checkbox("Checkbox");
+        Scrollbar scrollbar = new Scrollbar(Scrollbar.HORIZONTAL, 0, 1, 0, 255);
+        TextField textfield = new TextField(15);
+        TextArea textarea = new TextArea(5, 15);
+
+        components = new Component[] { panel, button, label, list, choice,
+            checkbox, scrollbar, textfield, textarea, frame };
+
+        for (int i = 0; i < components.length - 1; i++) {
+            components[i].addComponentListener(componentListener);
+            frame.add(components[i]);
+        }
+        frame.addComponentListener(componentListener);
+
+        frame.pack();
+        frame.setLocationRelativeTo(null);
+        frame.setVisible(true);
+    }
+
+    public static void main(String[] args) throws Exception {
+        try {
+            robot = new Robot();
+            robot.setAutoWaitForIdle(true);
+
+            EventQueue.invokeAndWait(ComponentEventTest::initializeGUI);
+            robot.waitForIdle();
+            robot.delay(DELAY);
+
+            doTest();
+
+            System.out.println("Test PASSED");
+        } finally {
+            EventQueue.invokeAndWait(ComponentEventTest::disposeFrame);
+        }
+    }
+
+    private static void doTest()
+        throws InvocationTargetException, InterruptedException {
+
+        EventQueue.invokeAndWait(() -> {
+            compAt = components[9].getLocationOnScreen();
+            compSize = components[9].getSize();
+        });
+
+        robot.mouseMove(compAt.x + compSize.width / 2,
+            compAt.y + compSize.height / 2);
+
+        robot.mousePress(InputEvent.BUTTON1_DOWN_MASK);
+        robot.mouseRelease(InputEvent.BUTTON1_DOWN_MASK);
+
+        for (int i = 0; i < components.length; i++) {
+            for (boolean state : new boolean[] { true, false }) {
+                doTest(i, state);
+            }
+        }
+
+        robot.delay(DELAY);
+
+        resetValues();
+        EventQueue.invokeAndWait(() -> frame.setExtendedState(Frame.ICONIFIED));
+
+        robot.delay(DELAY);
+        if (componentShown || componentHidden || componentMoved
+            || componentResized) {
+            printErrEvents("ComponentEvent triggered when frame is iconified");
+        }
+
+        resetValues();
+        EventQueue.invokeAndWait(() -> frame.setExtendedState(Frame.NORMAL));
+
+        robot.delay(DELAY);
+        if (componentShown || componentHidden) {
+            printErrEvents("ComponentEvent triggered when frame set to normal");
+        }
+    }
+
+    private static void printErrEvents(String errorMsg) {
+        System.err.print("Events triggered are: ");
+        synchronized (events) {
+            for (ComponentEvent event : events) {
+                System.err.print(event + "; ");
+            }
+        }
+        System.err.println();
+        throw new RuntimeException("FAIL: " + errorMsg);
+    }
```

Comment:
So you're collecting a list of events to print them if iconify or deiconify generates an unexpected list of events.

In all other cases `events` isn't used at all.

Given that all the generated events are printed as they're come, I see no reason why `events` is needed at all. By printing messages that you're about to test iconifying frame and deiconifying frame you separate the cases, the list of events will be printed between these messages.

Reply:
removed the events  , now list of events will be printed by using ComponentListener